Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculation.kt

```diff
@@ -1,17 +1,105 @@
 package de.rki.coronawarnapp.ccl.dccwalletinfo.calculation
 
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.gson.Gson
 import de.rki.coronawarnapp.ccl.configuration.model.CCLConfiguration
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.CclCertificate
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.Cose
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.Cwt
 import de.rki.coronawarnapp.ccl.dccwalletinfo.model.DccWalletInfo
-import de.rki.coronawarnapp.ccl.dccwalletinfo.model.dummyDccWalletInfo
-import de.rki.coronawarnapp.covidcertificate.common.certificate.DccData
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.DccWalletInfoInput
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.SystemTime
+import de.rki.coronawarnapp.covidcertificate.common.certificate.CwaCovidCertificate
+import de.rki.coronawarnapp.covidcertificate.validation.core.rule.DccValidationRule
+import de.rki.coronawarnapp.util.serialization.BaseGson
+import de.rki.jfn.JsonFunctions
 import javax.inject.Inject
 
-class DccWalletInfoCalculation @Inject constructor() {
-    fun getDccWalletInfo(
+class DccWalletInfoCalculation @Inject constructor(
+    @BaseGson val gson: Gson
+) {
+
+    private lateinit var jsonFunctions: JsonFunctions
+    private lateinit var boosterRulesNode: JsonNode
+
+    private val mapper by lazy { ObjectMapper() }
+
+    fun init(
         cclConfiguration: CCLConfiguration,
-        dccList: List<DccData<*>>
+        boosterRules: List<DccValidationRule>
+    ) {
+        jsonFunctions = JsonFunctions()
+        boosterRulesNode = Gson().toJson(boosterRules).toJsonNode()
+        cclConfiguration.logic.jfnDescriptors.forEach {
+            jsonFunctions.registerFunction(it.name, it.definition.toJsonNode())
+        }
+    }
+
+    fun getDccWalletInfo(
+        dccList: List<CwaCovidCertificate>
     ): DccWalletInfo {
+
+        val input = getDccWalletInfoInput(dccList).toJsonNode()
+        val output = jsonFunctions.evaluateFunction(
+            FUNCTION_NAME,
+            input
+        )
+
+        return mapper.treeToValue(output, DccWalletInfo::class.java)
```

Comment:
`mapper.readValue` ? it will parse to the right type

Reply:
compiler says no